lib/utilities/reportHandler.js

```diff
@@ -76,19 +76,38 @@ function getSystemStats() {
 
 function getCRRStats(log, cb) {
     log.debug('getting CRR stats', { method: 'getCRRStats' });
-    // TODO: Reuse metrics code from Backbeat by moving it to Arsenal instead of
-    // making an HTTP request to the Backbeat metrics route.
-    const { host, port } = config.backbeat;
-    const params = { url: `http://${host}:${port}/_/metrics/crr/all` };
-    return request.get(params, (err, res) => {
+    const { replicationEndpoints, redis } = config;
+    if (!redis) {
+        log.debug('redis connection not configured', { method: 'getCRRStats' });
+        return process.nextTick(() => cb(null, {}));
+    }
+    const sites = replicationEndpoints.map(endpoint => endpoint.site);
+    const backbeatMetrics = new backbeat.Metrics({
+        redisConfig: redis,
+        validSites: sites,
+        internalStart: Date.now() - 900000, // 15 minutes ago.
+    }, log);
+    const redisKeys = {
+        ops: 'bb:crr:ops',
+        bytes: 'bb:crr:bytes',
+        opsDone: 'bb:crr:opsdone',
+        bytesDone: 'bb:crr:bytesdone',
+        failedCRR: 'bb:crr:failed',
+    };
+    const routes = backbeat.routes(redisKeys, sites);
```

Comment:
As a future cleanup, I'm thinking the `redisKeys` and `sites` filtering should not be processed by the routes selection function, but by backend methods.

Reply:
I didn't like this option much either, but was felt that avoiding it would lead to a significantly bigger refactor. I agree this could be another round of work.